source/python/idetect/worker.py

```diff
@@ -45,29 +46,35 @@ def work(self):
             article = None
             while not claimed_one:
                 try:
-                    article = self.session.query(Article).filter(Article.status == self.status) \
-                        .order_by(Article.updated).first()  # choose the least-recently-updated article with this status
+                    # Only consider the most recent article for each url_id
+                    # ... that has the right status
+                    # ... sort by updated date
+                    # ... pick the first (oldest)
+                    article = Article.select_latest_version(self.session) \
+                        .filter(Article.status == self.status) \
+                        .order_by(Article.updated) \
+                        .first()
                     if article is None:
                         return False  # no work to be done
-                    article.update_status(self.working_status)
+                    article.create_new_version(self.working_status)
                     self.session.commit()
```

Comment:
Right. To create an edit history for the Article, we're updating the version whenever it's changed.